From 62f02d841ea2fb19e365af01892e793ed923ce06 Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
Date: Sat, 19 Apr 2008 17:57:09 +0200
Subject: [PATCH] __stack_chk_fail_local function

Added __stack_chk_fail_local() function as an alias for __stack_chk_fail().
This is required for dynamic linking.
---
 lib/stack_smash_handler2.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/lib/stack_smash_handler2.c b/lib/stack_smash_handler2.c
index ecefeb8..b03c0d5 100644
--- a/lib/stack_smash_handler2.c
+++ b/lib/stack_smash_handler2.c
@@ -2,6 +2,7 @@
 #include <write12.h>
 
 void __stack_chk_fail(void);
+void __stack_chk_fail_local(void) __attribute__((__alias__("__stack_chk_fail")));
 
 /* earlier versions of ProPolice actually gave the address and function
  * name as arguments to the handler, so it could print some useful
-- 
1.6.0.6

